lib/contentProfileListener.js

```diff
@@ -128,25 +128,25 @@ class ContentProfileImporter extends SHRContentProfileParserListener {
             const el = this._specs.dataElements.findByIdentifier(id);
             [el.value, ...el.fields].forEach(field => {
               const key = (field && field.identifier) ? field.identifier : 'value';
-              if (field && !fields.get(key)) fields.set(key, field);
+              if (field && !fields.has(key)) fields.set(key, field);
             });
           }
-          
-          let value = Array.from(fields.values()).find(field => {
-            let identifier;
 
+          let value = Array.from(fields.values()).find(field => {
             if (field instanceof IdentifiableValue) {
-              identifier = field.possibleIdentifiers.find(id => name === id.name);
+              // match name on effectiveIdentifier since CP requires author to use constrained type name
+              return field.effectiveIdentifier.name === name;
             } else if (field instanceof ChoiceValue) {
-              identifier = field.aggregateOptions.find(id => name === id.name);
+              // match name on one of the choice option's effectiveIdentifier
+              return field.aggregateOptions.some(o => o.effectiveIdentifier && o.effectiveIdentifier.name === name);
             }
 
-            return (identifier != null);
+            return false;
           });
 
           let element;
           if (value && value.identifier) {
```

Comment:
Should this check be `value && value.effectiveIdentifier` now given the below changes, or am I missing something?

Reply:
It's fine.  If it has an `identifier` then it is guaranteed to have an `effectiveIdentifier`.  And there isn't any case where it would have have an `effectiveIdentifier` but not an `identifier`.

That said, if it might be confusing to others, I'm glad to change it.

Reply:
OK.  I just pushed a change.